Lib/subprocess.py

```diff
@@ -412,6 +422,24 @@ def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
             stdout, stderr = process.communicate()
             raise TimeoutExpired(process.args, timeout, output=stdout,
                                  stderr=stderr)
+        except KeyboardInterrupt:
```

Comment:
I dislike the idea of having a special case only for KeyboardInterrupt. I would prefer to put this code in the bare "except:" case.